easybuild/main.py

```diff
@@ -199,6 +200,10 @@ def main(args=None, logfile=None, do_build=None, testing=False, modtool=None):
     eb_go, cfg_settings = set_up_configuration(args=args, logfile=logfile, testing=testing)
     options, orig_paths = eb_go.options, eb_go.args
 
+    if options.include_easyblocks_from_pr:
+        included_easyblocks = fetch_easyblocks_from_pr(options.include_easyblocks_from_pr)
```

Comment:
We rely heavily on `build_option` in `tools/github.py`, so only using `fetch_easyblocks_from_pr` after setting up the configuration indeed makes most sense.

I would move this inside of `set_up_configuration` though, since it's really part of setting up the configuration?
As long as you do it after `init_build_options`, you're good.
Do add a comment to explain why `include_easyblocks` is called there too though, and not in `_postprocess_include`.

That way people who leverage EasyBuild as a library (which requires calling `set_up_configuration` first) can also pull in easyblocks from a pull requests (by defining `$EASYBUILD_INCLUDE_EASYBLOCKS_FROM_PR`)...

Reply:
We rely heavily on `build_option` in `tools/github.py`, so only using `fetch_easyblocks_from_pr` after setting up the configuration indeed makes most sense.

I would move this inside of `set_up_configuration` though, since it's really part of setting up the configuration?
As long as you do it after `init_build_options`, you're good.
Do add a comment to explain why `include_easyblocks` is called there too though, and not in `_postprocess_include`.

That way people who leverage EasyBuild as a library (which requires calling `set_up_configuration` first) can also pull in easyblocks from a pull requests (by defining `$EASYBUILD_INCLUDE_EASYBLOCKS_FROM_PR`)...